### PR TITLE
Remove fomantic transition module

### LINKED ISSUE #15709: [Feature Request] option to disable animations
Hello,
Gitea has almost everywhere animations: the "Create" menu, the issue preview...
The problem is, I don't like these animations, I think they're a bit disturbing. So, I think it would be nice if there's an option in the settings to disable these animations.
Thanks!

Can you reproduce the bug at https://try.gitea.io:
  - [x] Yes
  - [ ] No

### PR DESCRIPTION
**Pull request by silverwind**
Removes all dropdown and dimmer animations. Works everywhere as far as I can tell, but need to give this thorough testing. Removes around 70kb JS/CSS.

Note, I'm not 100% sure regarding the various callbacks, those will need more investigation, but it appears to work nonetheless.

Fixes: https://github.com/go-gitea/gitea/issues/15709

### CODE REVIEW

**Comment by silverwind** [comment]
Has bugs with menu hiding in the navbar, wip.

**Comment by silverwind** [comment]
Hiding bug fixed. Fomantic apparently never removes the `transition` class after transition, so I replicated that and that makes the hiding CSS work again.

File: web_src/js/modules/fomantic.js [hunk 1]
```diff
@@ -22,6 +22,41 @@ export function initGiteaFomantic() {
...
+
+    const isIn = arg?.animation?.endsWith(' in');
+    const isOut = arg?.animation?.endsWith(' out');
+
+    let ret;
+    if (arg === 'show' || isIn) {
+      arg?.onStart?.(this);
+      ret = this.each((_, el) => {
+        $(el).removeClass('hidden');
+        $(el).addClass('visible');
```
Comment (delvh) [inline web_src/js/modules/fomantic.js]: I guess there's no way to use our own mechanism with `gt-hidden`, correct?

File: web_src/js/modules/fomantic.js [hunk 2]
```diff
@@ -22,6 +22,41 @@ export function initGiteaFomantic() {
...
+        if (isIn) $(el).addClass('transition');
+        arg?.onShow?.(this);
+      });
+      arg?.onComplete?.(this);
+    } else if (arg === 'hide' || isOut) {
+      arg?.onStart?.(this);
+      ret = this.each((_, el) => {
+        $(el).addClass('hidden');
+        $(el).removeClass('visible');
```
Comment (delvh) [inline web_src/js/modules/fomantic.js]: Same as above.

**Review by delvh — APPROVED** [review]

**Comment by wxiaoguang** [comment]
It breaks the modal dialogs.

File: web_src/js/modules/fomantic.js (hunk 1 above)
Comment (wxiaoguang) [reply web_src/js/modules/fomantic.js]: No, they are different. Fomantic always use "hidden" internally.

Comment (silverwind) [reply web_src/js/modules/fomantic.js]: No, other fomantic code relies on these classes.

**Comment by silverwind** [comment]
I see what you mean, the position is incorrect, but modal works otherwise. This was not an issue in an earlier iteration of this, checking...

Screenshot 2023-08-14 at 15 01 49

**Comment by silverwind** [comment]
Problem is the `display: flex !important;` inline style is missing, but interestingly, the [module](https://github.com/fomantic/Fomantic-UI/blob/develop/src/definitions/modules/transition.js) does not contain `flex` in the source.

**Comment by silverwind** [comment]
Found it:

https://github.com/fomantic/Fomantic-UI/blob/de758dcb3b4b51657623d43893bc5351aadaf942/src/definitions/modules/transition.js#L236-L239

**Comment by silverwind** [comment]
Modal should be fixed. This replacement function is likely not 100% equivalent to the removed module, but it appears to work well enough.

**Comment by wxiaoguang** [comment]
TBH, I am not a fan of removing all animations.

According to modern web sites:

1. For dropdowns, it doesn't need to use animation in most cases
2. For fly-in and modal dialogs, in most cases, they use animations, I have checked Apple, Google


So I do not think it's right to remove the animations totally.

**Comment by silverwind** [comment]
I like them nowhere and selectively allowing some would be a much more deeper dive into the hacky internals of Fomantic that I'm not interested in doing.

**Comment by silverwind** [comment]
Checked how GitHub does it, they have a 300ms `animation` on the modal content, but not the backdrop, which shows instantly. Hide is instant on both backdrop and modal.

**Comment by silverwind** [comment]
Modal now fades in. I also added a new CSS variable for the backdrop color and very slightly reduced it's opacity. I think it will be useful for themes.

Screenshot 2023-08-15 at 15 21 30
Screenshot 2023-08-15 at 15 21 41

**Comment by silverwind** [comment]
One more tweak done: added box shadows to dropdowns so they stand out more against the background:

Screenshot 2023-08-16 at 00 36 24

Screenshot 2023-08-16 at 00 36 12

Screenshot 2023-08-16 at 00 26 25
Screenshot 2023-08-16 at 00 26 14

(subpixel misalignment in second screenshot is a fomantic bug)
also, fixed an issue with the border color changing unexpectedly on focus change.

**Review by wxiaoguang — APPROVED** [review]
Haven't tested, while it might be worth to try.

**Pull request merged**